skyportal/handlers/api/comment.py

```diff
@@ -174,69 +182,80 @@ def get(self, associated_resource_type, resource_id, comment_id=None):
-                        .unique()
-                        .all()
+                if resource_id is None and (text is None or str(text).strip() == ""):
+                    return self.error(
+                        "Please provide a resource_id or text to search for."
                     )
+                table, resource_id_col = None, None
+                if associated_resource_type.lower() == "sources":
+                    table, resource_id_col = Comment, "obj_id"
                 elif associated_resource_type.lower() == "spectra":
-                    comments = (
-                        session.scalars(
-                            CommentOnSpectrum.select(self.current_user).where(
-                                CommentOnSpectrum.spectrum_id == resource_id
-                            )
-                        )
-                        .unique()
-                        .all()
-                    )
+                    table, resource_id_col = CommentOnSpectrum, "spectrum_id"
                 elif associated_resource_type.lower() == "gcn_event":
-                    comments = (
-                        session.scalars(
-                            CommentOnGCN.select(self.current_user).where(
-                                CommentOnGCN.gcn_id == resource_id
-                            )
-                        )
-                        .unique()
-                        .all()
-                    )
+                    table, resource_id_col = CommentOnGCN, "gcn_id"
                 elif associated_resource_type.lower() == "earthquake":
-                    comments = (
-                        session.scalars(
-                            CommentOnEarthquake.select(self.current_user).where(
-                                CommentOnEarthquake.earthquake_id == resource_id
-                            )
-                        )
-                        .unique()
-                        .all()
-                    )
+                    table, resource_id_col = CommentOnEarthquake, "earthquake_id"
                 elif associated_resource_type.lower() == "shift":
-                    comments = (
-                        session.scalars(
-                            CommentOnShift.select(self.current_user).where(
-                                CommentOnShift.shift_id == resource_id
-                            )
-                        )
-                        .unique()
-                        .all()
-                    )
+                    table, resource_id_col = CommentOnShift, "shift_id"
                 else:
                     return self.error(
                         f'Unsupported associated resource type "{associated_resource_type}".'
                     )
 
-                query_output = [
-                    {**c.to_dict(), 'resourceType': associated_resource_type.lower()}
-                    for c in comments
-                ]
+                stmt = table.select(self.current_user)
```

Comment:
This should be associated_user_or_id or whatever we use in case of tokens I think.

Reply:
good catch, new code inherited from incorrect old code 😉

Reply:
done, also updated it everywhere else in that file

Reply:
@Theodlz great, eventually we will have squashed them ;)